src/pages/signin/SignInPage.js

```diff
@@ -154,16 +171,18 @@ class SignInPage extends Component {
                 <SignInPageLayout
                     welcomeHeader={welcomeHeader}
                     welcomeText={welcomeText}
-                    shouldShowWelcomeHeader={showLoginForm || showPasswordForm || showValidateCodeForm || showUnlinkLoginForm || !this.props.isSmallScreenWidth}
-                    shouldShowWelcomeText={showLoginForm || showPasswordForm || showValidateCodeForm}
+                    shouldShowWelcomeHeader={showLoginForm || showPasswordForm || showValidateCodeForm || showUnlinkLoginForm || showWelcomeForm || !this.props.isSmallScreenWidth}
+                    shouldShowWelcomeText={showLoginForm || showPasswordForm || showValidateCodeForm || showWelcomeForm}
                 >
                     {/* LoginForm and PasswordForm must use the isVisible prop. This keeps them mounted, but visually hidden
                     so that password managers can access the values. Conditionally rendering these components will break this feature. */}
                     <LoginForm
                         isVisible={showLoginForm}
                         blurOnSubmit={this.props.account.validated === false}
                     />
-                    {showValidateCodeForm ? <ValidateCodeForm isVisible={showValidateCodeForm} /> : <PasswordForm isVisible={showPasswordForm} />}
+                    {showWelcomeForm && <WelcomeForm />}
+                    {showValidateCodeForm && <ValidateCodeForm isVisible={showValidateCodeForm} />}
+                    {showPasswordForm && <PasswordForm isVisible={showPasswordForm} />}
```

Comment:
The props `isVisible` look a bit useless at this point. `isVisible` will be always `true` if we are rendering these components.

Reply:
```
{showValidateCodeForm ? <ValidateCodeForm isVisible={showValidateCodeForm} /> : <PasswordForm isVisible={showPasswordForm} />}
```

@aldo-expensify, that's true. That was already there from the previous code.

I can create [another issue](https://github.com/Expensify/App/issues/19787) for removing the unnecessary prop from `ValidateCodeForm` (I assume `PasswordForm` will be removed soon anyway)

Reply:
that sounds good with me